test/e2e/opaque_resource.go

```diff
@@ -0,0 +1,309 @@
+		awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+			rQuant, found := n.Status.Capacity[opaqueResName]
+			return found && rQuant.MilliValue() == int64(5000)
+		})
+
+		By("Watching node status to verify the advertised opaque resource is eventually present in allocatable")
+		awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+			rQuant := n.Status.Allocatable[opaqueResName]
+			return rQuant.MilliValue() == int64(5000)
+		})
+	})
+
+	Context("With opaque resources already advertised", func() {
+		BeforeEach(func() {
+			node = refresh(f, node)
+			// Use the client to patch node capacity, adding opaque "foo" resource.
+			patch := []byte(`[{"op": "add", "path": "/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-foo", "value": "5"}]`)
+			f.Client.Patch(api.JSONPatchType).Resource("nodes").Name(node.Name).SubResource("status").Body(patch).Do()
+			awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+				rQuant, found := n.Status.Allocatable[opaqueResName]
+				return found && rQuant.MilliValue() == int64(5000)
+			})
+			node = refresh(f, node)
+		})
+
+		It("should not break pods that do not consume opaque integer resources.", func() {
+			By("Verifying that a vanilla pod can still be scheduled")
+			requests := api.ResourceList{api.ResourceCPU: resource.MustParse("0.1")}
+			limits := api.ResourceList{api.ResourceCPU: resource.MustParse("0.2")}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "without-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pod).To(beScheduled())
+		})
+
+		It("should schedule pods that do consume opaque integer resources.", func() {
+			By("Creating a pod that requires less of the opaque resource than is allocatable on a node.")
+			requests := api.ResourceList{
+				api.ResourceCPU: resource.MustParse("0.1"),
+				opaqueResName:   resource.MustParse("1"),
+			}
+			limits := api.ResourceList{
+				api.ResourceCPU: resource.MustParse("0.2"),
+				opaqueResName:   resource.MustParse("2"),
+			}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "min-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pod).To(beScheduled())
+		})
+
+		It("should not schedule pods that exceed the available amount of opaque integer resource.", func() {
+			By("Creating a pod that requires more of the opaque resource than is allocatable on any node")
+			requests := api.ResourceList{opaqueResName: resource.MustParse("6")}
+			limits := api.ResourceList{}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "over-max-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			// TODO(CD): Watch scheduler events instead to catch an indication of
+			//           insufficient resources or other unschedulable condition
+			//           instead of waiting for this to time out.
+			Expect(pod).To(notBeScheduled())
```

Comment:
Sorry, the TODO is out-dated. I asked around in sig-testing for recommendations and the only feedback I got was not to rely on events. This doesn't wait for the `beScheduled` predicate to time out, rather it looks for the unschedulable condition on the pod.

I'm going to delete the comment here because it's not relevant. If you would still like these to be refactored in light of the above please let me know.